packages/types/api/providers.d.ts

```diff
@@ -0,0 +1,32 @@
+export interface ProviderRecord {
+  id: string,
+  globalConnectionLimit?: number,
+  protocol: string,
+  host: string,
+  port?: number,
+  username?: string,
+  password?: string,
+  encrypted?: boolean,
+  createdAt: number,
+  updatedAt?: number,
+  privateKey?: string,
+  cmKeyId?: string,
+  certificateUri?: string
+}
+
+export interface PostgresProviderRecord {
+  cumulusId?: number,
+  name: string,
+  globalConnectionLimit?: number,
+  protocol: string,
+  host: string,
+  port?: number,
+  username?: string,
+  password?: string,
+  created_at?: number,
+  updated_at?: number,
+  privateKey?: string,
+  cmKeyId?: string,
+  certificateUri?: string
+  [key: string]: any
```

Comment:
What is this for?

Reply:
Sorry I missed this comment -- it's a type index, it allowed assignment via undeclared `obj[string]` and other side effects (some not good).  I found a better solution to that in: 

c591c12ea